ubuntu/small-ubuntu-24.04.pkr.hcl

```diff
@@ -21,41 +21,38 @@ variable "SSH_PUB_KEY" {
   default = ""
 }
 
-source "qemu" "small_ubuntu_20_04" {
+source "qemu" "small_ubuntu_24_04" {
   boot_command              = [
-    "<enter><enter><f6><esc><wait>",
-    "<bs><bs><bs><bs><bs><bs><bs><bs>",
-    "<bs><bs><bs><bs><bs><bs><bs><bs>",
-    "<bs><bs><bs><bs><bs><bs><bs><bs>",
-    "<bs><bs><bs><bs><bs><bs><bs><bs>",
-    "<bs><bs><bs><bs><bs><bs><bs><bs>",
-    "/casper/vmlinuz ",
-    "initrd=/casper/initrd ",
-    " autoinstall ds=nocloud-net;s=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ ",
+    "c<wait>",
+    "linux /casper/vmlinuz --- autoinstall ds=\"nocloud-net;seedfrom=http://{{ .HTTPIP }}:{{ .HTTPPort }}/\"",
     " PACKER_USER=ubuntu PACKER_AUTHORIZED_KEY={{ `${var.SSH_PUB_KEY}` | urlquery }}",
-    "<wait><enter>"
+    "<enter><wait>",
+    "initrd /casper/initrd",
+    "<enter><wait>",
+    "boot",
+    "<enter>"
   ]
-  boot_wait                 = "3s"
-  disk_size                 = 3500
+  boot_wait                 = "6s"
+  disk_size                 = 4000
   format                    = "qcow2"
   headless                  = true
   http_directory            = "httpdir"
   http_port_max             = 8550
   http_port_min             = 8500
-  iso_checksum              = "sha256:b8f31413336b9393ad5d8ef0282717b2ab19f007df2e9ed5196c13d8f9153c8b"
-  iso_url                   = "https://releases.ubuntu.com/20.04/ubuntu-20.04.6-live-server-amd64.iso"
+  iso_checksum              = "sha256:d6dab0c3a657988501b4bd76f1297c053df710e06e0c3aece60dead24f270b4d"
```

Comment:
Consider whether this value cannot be declared from a data source via an HTTP call:
https://developer.hashicorp.com/packer/docs/datasources/http

If the shasum is published, retrieve and extract it from a page.